### PR TITLE
more clang-tidy fixes

### PR DESCRIPTION
**Pull request by neheb**
- [x] Code style update

I wonder if the last one is a bugfix.

### CODE REVIEW

**Comment by antony-jr** [comment]
You might want to remove this https://github.com/antony-jr/QArchive/pull/85/commits/b2d63046f24c9e8c653f859a94170ac24953ae3f commit, it's causing a conflict. I merged the other PR which adds this change. You can pull from master, rebase and then force push fixing the conflict.

File: include/qarchivememoryextractoroutput.hpp [hunk 1]
```diff
@@ -1,24 +1,23 @@
...
 class QARCHIVE_EXPORT MemoryExtractorOutput : public QObject {
     using QObject::QObject;
     Q_OBJECT
-  public:
-    MemoryExtractorOutput(std::unique_ptr<QVector<MemoryFile>>, QObject *parent = nullptr);
+public:
+    explicit MemoryExtractorOutput(std::unique_ptr<std::vector<MemoryFile>>, QObject* parent = nullptr);
 
-    [[gnu::warn_unused_result]] const QVector<MemoryFile> &getFiles() const;
+    [[gnu::warn_unused_result]] const std::vector<MemoryFile>& getFiles() const;
```
Comment (antony-jr) [inline include/qarchivememoryextractoroutput.hpp:17]: This breaks the public API, we might want to keep that for the next release. Also is it bad using QVector? for now I think we will have a method like ```getFilesStdVector()``` for this implementation, we might remove the QVector version in the next release and rename this.

Comment (neheb) [reply include/qarchivememoryextractoroutput.hpp:17]: Good point. std::vector allows using emplace_back. append() for some strange reason requires a defaulted constructor.

**Comment by neheb** [comment]
```
                 from /home/runner/work/QArchive/QArchive/src/qarchiveextractor_p.cc:6:
/home/runner/work/QArchive/QArchive/Qt/Qt/5.12.12/gcc_64/include/QtCore/qvector.h: In instantiation of ‘void QVector<T>::reallocData(int, int, QArrayData::AllocationOptions) [with T = QArchive::MemoryFile; QArrayData::AllocationOptions = QFlags<QArrayData::AllocationOption>]’:
/home/runner/work/QArchive/QArchive/Qt/Qt/5.12.12/gcc_64/include/QtCore/qvector.h:696:9:   required from ‘void QVector<T>::append(T&&) [with T = QArchive::MemoryFile]’
/home/runner/work/QArchive/QArchive/Qt/Qt/5.12.12/gcc_64/include/QtCore/qvector.h:262:29:   required from ‘void QVector<T>::push_back(T&&) [with T = QArchive::MemoryFile]’
/home/runner/work/QArchive/QArchive/src/qarchiveextractor_p.cc:938:53:   required from here
/home/runner/work/QArchive/QArchive/Qt/Qt/5.12.12/gcc_64/include/QtCore/qvector.h:599:29: error: no matching function for call to ‘QArchive::MemoryFile::MemoryFile()’
  599 |                             new (dst++) T();
      |                             ^~~~~~~~~~~~~~~
In file included from /home/runner/work/QArchive/QArchive/include/qarchivememoryextractoroutput.hpp:9,
                 from /home/runner/work/QArchive/QArchive/include/qarchiveextractor_p.hpp:16,
                 from /home/runner/work/QArchive/QArchive/src/qarchiveextractor_p.cc:6:
/home/runner/work/QArchive/QArchive/include/qarchivememoryfile.hpp:13:5: note: candidate: ‘QArchive::MemoryFile::MemoryFile(QJsonObject, std::shared_ptr<QBuffer>)’
   13 |     MemoryFile(QJsonObject, std::shared_ptr<QBuffer>);
      |     ^~~~~~~~~~
/home/runner/work/QArchive/QArchive/include/qarchivememoryfile.hpp:13:5: note:   candidate expects 2 arguments, 0 provided
/home/runner/work/QArchive/QArchive/include/qarchivememoryfile.hpp:11:23: note: candidate: ‘QArchive::MemoryFile::MemoryFile(const QArchive::MemoryFile&)’
   11 | class QARCHIVE_EXPORT MemoryFile {
      |                       ^~~~~~~~~~
```

The fact that this happens with Qt5 and not Qt6 tells me this is a bug.

**Comment by neheb** [comment]
@antony-jr  Should I place the last 4 commits in a separate PR?

**Comment by antony-jr** [comment]
No problem. Looks fine. I will merge this soon.

**Pull request merged**